_P208_Nokia_LCD_5110.ino

```diff
@@ -0,0 +1,303 @@
+#define PLUGIN_VALUENAME3_208 "Rotation"
+
+#include <Adafruit_GFX.h>
+#include <Adafruit_PCD8544.h>
+#define lcd_lines 6
+#define digits_per_template_line 48
+#define digits_per_display_line 14
+int8_t aa = 2;
+int8_t ab = 3;
+
+Adafruit_PCD8544 *lcd3 = nullptr;
+
+char html_input [lcd_lines][digits_per_display_line];
+
+boolean Plugin_208(byte function, struct EventStruct *event, String& string){
+boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:{
+        Device[++deviceCount].Number = PLUGIN_ID_208 ;
+        Device[deviceCount].Type = DEVICE_TYPE_SINGLE;
+        Device[deviceCount].VType = Sensor_VType::SENSOR_TYPE_SINGLE;
+        Device[deviceCount].Ports = 0;
+        Device[deviceCount].PullUpOption = false;
+        Device[deviceCount].InverseLogicOption = false;
+        Device[deviceCount].FormulaOption = false;
+        Device[deviceCount].ValueCount = 3;
+        Device[deviceCount].SendDataOption = false;
+        Device[deviceCount].TimerOption = true;
+        Device[deviceCount].TimerOptional = true;
+        break;
+      }
+
+    case PLUGIN_GET_DEVICENAME:{
+        string = F(PLUGIN_NAME_208);
+        break;
+      }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:{
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_208));
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_208));
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[2], PSTR(PLUGIN_VALUENAME3_208));
+        break;
+      }
+
+    case PLUGIN_WEBFORM_LOAD:{
+        addFormNumericBox(F("Display Contrast(50-100):"), F("plugin_208_contrast"), Settings.TaskDevicePluginConfig[event->TaskIndex][1]);
+
+        byte choice8 = Settings.TaskDevicePluginConfig[event->TaskIndex][6];
+        byte choice9 = Settings.TaskDevicePluginConfig[event->TaskIndex][7];
+        addFormPinSelect(F("CE (LCD_pin_2)"), F("plugin_208_GPIO_CE"), choice8);
+        addFormPinSelect(F("DC (LCD_pin_3)"), F("plugin_208_GPIO_DC"), choice9);
+        addFormPinSelect(F("Backlight PIN"), F("taskdevicepin3"), Settings.TaskDevicePin3[event->TaskIndex]);
+
+        byte choice3 = Settings.TaskDevicePluginConfig[event->TaskIndex][2];
+        int optionValues3[4] = { 0, 1, 2, 3 };
+        String options3[4] = { F("0"), F("90"), F("180"), F("270") };
+        addFormSelector(F("Display Rotation"), F("plugin_208_rotation"), 4, options3, optionValues3, choice3);
+
+        byte choice4 = Settings.TaskDevicePluginConfig[event->TaskIndex][0];
+        String options4[2] = { F("OFF"), F("ON") };
+        int optionValues4[2] = { 0, 1 };
+        addFormSelector(F("Backlight"), F("plugin_208_backlight"), 2, options4, optionValues4, choice4);
+
+        byte choice5 = Settings.TaskDevicePluginConfig[event->TaskIndex][3];
+        byte choice6 = Settings.TaskDevicePluginConfig[event->TaskIndex][4];
+        byte choice7 = Settings.TaskDevicePluginConfig[event->TaskIndex][5];

```

Comment:
To improve readability, you could use the `PCONFIG(5)` macro instead of `Settings.TaskDevicePluginConfig[event->TaskIndex][5]` (and also for the other settings, there are many examples in the mega repository).

Reply:
Great. Changed in next commit.